geniza/corpus/management/commands/import_data.py

```diff
@@ -336,3 +372,153 @@ def log_creation(self, *objects):
                 object_repr=str(obj),
                 change_message=self.logentry_message,
                 action_flag=ADDITION)
+
+    def get_user(self, name, pgpid=None):
+        """Find a user account based on a provided name, using a simple cache.
+
+        If not found, tries to use first/last initials for lookup. If all else
+        fails, use the generic team account (TEAM_USERNAME).
+        """
+
+        # check the cache first
+        user = self.user_lookup.get(name)
+        if user:
+            logging.debug(
+                f"Using cached user {user} for {name} on PGPID {pgpid}")
+            return user
+
+        # person with given name(s) and last name – case-insensitive lookup
+        if " " in name:
+            given_names, last_name = [sname.strip(punctuation) for sname in
+                                      name.rsplit(" ", 1)]
+            try:
+                user = User.objects.get(first_name__iexact=given_names,
+                                        last_name__iexact=last_name)
+            except User.DoesNotExist:
+                pass
+
+        # initials; use first & last to do lookup
+        elif name:
+            first_i, last_i = name[0], name[-1]
+            try:
+                user = User.objects.get(first_name__startswith=first_i,
+                                        last_name__startswith=last_i)
+            except User.DoesNotExist:
+                pass
+
+        # if we didn't get anyone through either method, warn and use team user
+        if not user:
+            logging.warning(
+                f"Couldn't find user {name} on PGPID {pgpid}; using {self.team_user}")
+            return self.team_user
+
+        # otherwise add to the cache using requested name and return the user
+        logging.debug(f"Found user {user} for {name} on PGPID {pgpid}")
+        self.user_lookup[name] = user
+        return user
+
+    def get_edit_history(self, input_by, date_entered, pgpid=None):
+        """Parse spreadsheet "input by" and "date entered" columns to
+        reconstruct the edit history of a single document.
+
+        Output is a list of dict to pass to log_edit_history. Each event has
+        a type (creation or revision), associated user, and timestamp.
+        """
+
+        # split both fields by semicolon delimiter & remove whitespace
+        all_input = map(str.strip, input_by.split(";"))
+        all_dates = map(str.strip, date_entered.split(";"))
+
+        # try to map every "input by" listing to a user account
+        users = []
+        for input_by in all_input:
+
+            # special case: two people together; add as a tuple
+            if " and " in input_by:
+                users.append(
+                    tuple(map(self.get_user, input_by.split(" and "))))
+            else:
+                users.append(self.get_user(input_by))
```

Comment:
not sure what you mean – I need to keep the ones paired with "and" together so that they can be matched to a single timestamp, because it's technically a single event with two authors